Add yatemplate

mineo opened the pull request:
I wrote this package to use yasnippet for file templates. It uses auto-insert-mode to insert templates into new, empty files and then expands those with yasnippet. The direct link is https://github.com/mineo/yatemplate

purcell commented:
Thanks. You're declaring `lexical-binding`, perhaps simply because it's in the `auto-insert-mode` template, but it implies that the code will work only with Emacs 24 and greater. Please either remove that declaration, or add a package dependency on `(emacs "24")`.

Additionally, the `Commentary:` section of the `.el` file doesn't describe how to use the package: this would be helpful information to add.

mineo commented:
I've removed `lexical-binding` and updated the commentary a bit - I hope I've made it more clear that all this package does is insert some elements into `auto-insert-mode`s `auto-insert-alist` and lets `auto-insert-mode` and `yasnippet` do all the heavy work.

purcell commented:
Super, thanks.
